src/frontend/planner_test/tests/testdata/basic_query.yaml

```diff
@@ -22,11 +22,11 @@
     select * from t where 1>2 and 1=1 and 3<1 and 4<>1 or 1=1 and 2>=1 and 1<=2;
   batch_plan: |
     BatchExchange { order: [], dist: Single }
-    └─BatchFilter { predicate: (1:Int32 = 1:Int32) AND ((((1:Int32 > 2:Int32) AND (3:Int32 < 1:Int32)) AND (4:Int32 <> 1:Int32)) OR ((2:Int32 >= 1:Int32) AND (1:Int32 <= 2:Int32))) }
+    └─BatchFilter { predicate: true:Boolean AND true:Boolean }
```

Comment:
That is because condition contains conjunctive expressions that are not folded. We can apply the reduction on condition after expressions are rewritten to only retain non-True predicates.

Reply:
Make sense. The predicate is a `Vec<Expr>` of conjunctions rather than a single expr. So this is expected rather than a bug at some unknown places. We can add this optimization separately (maybe same phase but also handles `Condition(Vec<Expr>)` as an expr).

Reply:
Semantically, a `Condition` with an expression `True` is as good as it without that expression. So I think folding the expression after rewriting it in the same phase is good enough.